### PR TITLE
Add Battle Frontier to Hoenn

### PR DESCRIPTION
**Pull request by apple096**
TODO:
- [ ] Finalize rewards
- [x] Show correct Pokemon Attack underneath Battle Frontier View
- [x] Hide Click Attack underneath Battle Frontier View : Made the Click Attack 0 in the Battle Frontier, instead of hiding it completely
- [x] Update layout of Bottom half of Battle Frontier : Added balls

### CODE REVIEW

**Comment by apple096** [comment]
https://cdn.discordapp.com/attachments/738744988883484742/739871838716559390/bff.png An example of the new layout

**Comment by RedSparr0w** [comment]
Nice! looks good.
What does it show for the non item based rewards?

File: src/index.html [hunk 1]
```diff
@@ -354,6 +367,51 @@ <h4><strong data-bind="text: DungeonRunner.timeLeftSeconds() + 's'"></strong></h
                 </div>
             </div>
             <!-- /ko -->
+
+            <!-- ko if: App.game.gameState === GameConstants.GameState.battleFrontier -->
+            <div id="battleFrontierInformation" class="card sortable-disabled border-secondary mb-3 battle-frontier no-select">
+                <div class="card-header p-0">
+                    <span>Battle Frontier (BETA)</span>
```
Comment (Aegyo) [inline src/index.html]: Do we want to remove the (BETA) before merging?

File: src/index.html [hunk 2]
```diff
@@ -333,6 +338,14 @@ <h4><strong data-bind="text: DungeonRunner.timeLeftSeconds() + 's'"></strong></h
                                   <span data-bind="text: App.game.party.calculateClickAttack().toLocaleString('en-US')"></span>
                               </span>
                             </td>
+                            <!-- /ko -->
+                            <!-- ko if: App.game.gameState === GameConstants.GameState.battleFrontier -->
+                            <td width="100%">
+                                <span style="display: inline;">Pokémon Attack:
+                                    <span data-bind="text: App.game.party.calculatePokemonAttack(PokemonType.None, PokemonType.None, true).toLocaleString('en-US');"></span>
```
Comment (Aegyo) [inline src/index.html:346]: Is this (None, None) on purpose, or leftover from an earlier testing phase?

File: src/index.html [hunk 3]
```diff
@@ -354,6 +367,51 @@ <h4><strong data-bind="text: DungeonRunner.timeLeftSeconds() + 's'"></strong></h
                 </div>
             </div>
             <!-- /ko -->
+
+            <!-- ko if: App.game.gameState === GameConstants.GameState.battleFrontier -->
+            <div id="battleFrontierInformation" class="card sortable-disabled border-secondary mb-3 battle-frontier no-select">
```
Comment (Aegyo) [inline src/index.html]: In the interest of keeping `index.html` from becoming a behemoth, we might consider moving this into another file and `@import`ing it

File: src/scripts/battleFrontier/BattleFrontierBattle.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        const enemy = pokemonMap.random(GameConstants.TotalPokemonsPerRegion[player.highestRegion()]);
+        const health = PokemonFactory.routeHealth(BattleFrontierRunner.stage() + 10, GameConstants.Region.kanto);
+        const level = Math.min(100, BattleFrontierRunner.stage());
+        // Don't award money per pokemon defeated, award money at the end
+        const money = 0;
+        const shiny = PokemonFactory.generateShiny(GameConstants.SHINY_CHANCE_BATTLE);
+        // Give 1 extra shard per pokemon defeated after every 80 stages
+        const shards = Math.ceil(BattleFrontierRunner.stage() / 80);
+
+        const enemyPokemon = new BattlePokemon(enemy.name, enemy.id, enemy.type[0] || PokemonType.None, enemy.type[1] || PokemonType.None, health, level, 0, enemy.exp, money, shiny, shards);
```
Comment (Aegyo) [inline src/scripts/battleFrontier/BattleFrontierBattle.ts]: I assume we are doing this `enemy.type[0] || PokemonType.None` everywhere. Seems like a good point to just make the BattlePokemon constructor default those values with `PokemonType.None`

File: src/styles/main.less [hunk 5]
```diff
@@ -465,6 +465,7 @@ img {
 
 *::-webkit-scrollbar-track
 {
+	box-shadow: inset 0 0 6px rgba(0,0,0,0.3);
```
Comment (Aegyo) [inline src/styles/main.less:468]: Not entirely sure it makes a difference, but shouldn't the vendor prefixed version go above this, so the browser uses the unprefixed version if it knows how?

File: src/index.html (hunk 2 above)
Comment (RedSparr0w) [reply src/index.html:346]: It's purposefully None None, this way it calculates the real damage, not based on damage to a specific type of pokemon

File: src/index.html (hunk 1 above)
Comment (apple096) [reply src/index.html]: Probably

File: src/styles/main.less (hunk 5 above)
Comment (RedSparr0w) [reply src/styles/main.less:468]: Doesn't look like there is any others types:
<https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar>
<https://www.w3schools.com/howto/howto_css_custom_scrollbar.asp>
Seems to be a webkit only thing

**Comment by Aegyo** [comment]
Changes look good :+1:

Is it just some better rewards needed on this now?

**Comment by apple096** [comment]
Yep

**Comment by RedSparr0w** [comment]
Merging into `feature/battleFrontier` branch, so we can keep track of future changes easier without having to review older changes into the branch.

**Pull request merged**